internal/validators/secretref/secret_ref.go

```diff
@@ -84,3 +91,101 @@ func checkForMissingFields(ref telemetryv1alpha1.SecretKeyRef) error {
 
 	return nil
 }
+
+func getSecretRefsTracePipeline(tp *telemetryv1alpha1.TracePipeline) []telemetryv1alpha1.SecretKeyRef {
+	return getRefsInOTLPOutput(tp.Spec.Output.OTLP)
+}
+
+func getSecretRefsMetricPipeline(mp *telemetryv1alpha1.MetricPipeline) []telemetryv1alpha1.SecretKeyRef {
+	return getRefsInOTLPOutput(mp.Spec.Output.OTLP)
+}
+
+func getSecretRefsLogPipeline(lp *telemetryv1alpha1.LogPipeline) []telemetryv1alpha1.SecretKeyRef {
+	var refs []telemetryv1alpha1.SecretKeyRef
+
+	for _, v := range lp.Spec.Variables {
+		if v.ValueFrom.SecretKeyRef != nil {
+			refs = append(refs, *v.ValueFrom.SecretKeyRef)
+		}
+	}
+
+	refs = append(refs, GetEnvSecretRefs(lp)...)
+	refs = append(refs, getTLSSecretRefs(lp)...)
+
+	return refs
+}
+
+// GetEnvSecretRefs returns the secret references of a LogPipeline that should be stored in the env secret
+func GetEnvSecretRefs(lp *telemetryv1alpha1.LogPipeline) []telemetryv1alpha1.SecretKeyRef {
```

Comment:
Does it have to be exported?

Reply:
I've re-implemented the logic in `internal/reconciler/logpipeline/fluentbit/sync.go`, avoiding the export of this function.